<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CurveNetMakerClass</class>
 <widget class="QMainWindow" name="CurveNetMakerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>CurveNetMaker</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="MyGLCanvas" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1600</width>
      <height>930</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>202</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QCheckBox" name="FeatureCheckBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>0</height>
        </size>
       </property>
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <property name="text">
        <string>Show Feature Line</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>10</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <widget class="QLabel" name="AlphaTxt">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Alpha </string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="AlphaSpin">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.050000000000000</double>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QCheckBox" name="SmoothBoundaryCheckBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Smooth Boundary</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1600</width>
     <height>18</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QMenu" name="menuOpen">
    <property name="title">
     <string>Open</string>
    </property>
    <addaction name="OpenMeshFile"/>
   </widget>
   <widget class="QMenu" name="menuSave">
    <property name="title">
     <string>Save</string>
    </property>
    <addaction name="SaveSegmentation"/>
   </widget>
   <addaction name="menuOpen"/>
   <addaction name="menuSave"/>
  </widget>
  <action name="OpenMesh">
   <property name="text">
    <string>Mesh File</string>
   </property>
  </action>
  <action name="SaveSegmentation">
   <property name="text">
    <string>Segmentation</string>
   </property>
  </action>
  <action name="OpenMeshFile">
   <property name="text">
    <string>Mesh</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLCanvas</class>
   <extends>QOpenGLWidget</extends>
   <header>myglcanvas.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="CurveNetMaker.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>OpenMeshFile</sender>
   <signal>triggered()</signal>
   <receiver>CurveNetMakerClass</receiver>
   <slot>OpenMeshFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>26</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SaveSegmentation</sender>
   <signal>triggered()</signal>
   <receiver>CurveNetMakerClass</receiver>
   <slot>SaveCurveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>74</x>
     <y>28</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AlphaSpin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CurveNetMakerClass</receiver>
   <slot>AlphaChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>995</y>
    </hint>
    <hint type="destinationlabel">
     <x>759</x>
     <y>972</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FeatureCheckBox</sender>
   <signal>clicked()</signal>
   <receiver>CurveNetMakerClass</receiver>
   <slot>ShowFeatureLines()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>570</x>
     <y>985</y>
    </hint>
    <hint type="destinationlabel">
     <x>434</x>
     <y>970</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SmoothBoundaryCheckBox</sender>
   <signal>clicked()</signal>
   <receiver>CurveNetMakerClass</receiver>
   <slot>SmoothBoundary()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1018</x>
     <y>976</y>
    </hint>
    <hint type="destinationlabel">
     <x>1066</x>
     <y>972</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OpenMeshFile()</slot>
  <slot>SaveCurveFile()</slot>
  <slot>AlphaChanged()</slot>
  <slot>ShowFeatureLines()</slot>
  <slot>StartInteraction()</slot>
  <slot>SmoothBoundary()</slot>
 </slots>
</ui>
